Data/BankerQueue/LazyLists.idr

```diff
@@ -0,0 +1,189 @@
+||| This is not actually a general lazy list library. It's really
+||| intended for implementing banker's queues. It also has some
+||| junk about regular lists for that purpose.
+module Data.BankerQueue.LazyLists
+
+%default total
+%access public
+
+||| A lazy-spined list.
+data LList : Type -> Type where
```

Comment:
No, `CoList` would not be appropriate here. If we used those, it would be impossible to convince the termination checker that we can pull all the elements out of the front list one at a time and eventually reach the end.